### PR TITLE
version refactor phase 1

### PR DESCRIPTION
**Pull request by saneshark**
Fixes https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1990

- removed a bunch of unnecessary methods.
- added a bunch of helpful methods
- ensured scopes always returning relation object

I'm calling this a phase 1, because I would still like to improve this more in the future, but seeing as we don't currently support rollback and other things that would increase complexity on this, i'm comfortable with just these changes.

### CODE REVIEW

**Review by mphprogrammer — APPROVED** [review]
LGTM - but we may want to change the partial that displays the preview record actions to use publish? rather than passing in a local. Thats in the other PR so lets push this one out, and then change the other pr (preview == production) to disable the publish button.

**Comment by saneshark** [comment]
Good point, i might as well change the query for fetching previews and production to use the updated scopes now that `newest` returns a relation object.

**Comment by saneshark** [comment]
@patrickvinograd are you good with me merging this one?

File: spec/controllers/dashboards_controller_spec.rb [hunk 1]
```diff
@@ -109,7 +109,7 @@ def load_table(klass, options)
         get :push
 
         expect(flash.alert).to eq('No preview version available')
-        expect(Version.production_version).to be_blank
+        expect(Version.current_production).to be_blank
```
Comment (patrickvinograd) [inline spec/controllers/dashboards_controller_spec.rb]: Should this be `current_preview`?

**Review by patrickvinograd — APPROVED** [review]
LGTM. One question on the dashboard spec, but I didn't read through the spec thoroughly, just noticed something potentially out of place.

**Comment by patrickvinograd** [comment]
I think we should leave it here in terms of refactoring, unless there's an actual bug fix needed.

File: spec/controllers/dashboards_controller_spec.rb (hunk 1 above)
Comment (saneshark) [inline spec/controllers/dashboards_controller_spec.rb]: the spec is testing to make sure that push returns a flash error and that nothing is pushed to production. The last expectation in the example is kind of dumb, but I guess it doesn't hurt to have it.

**Comment by saneshark** [comment]
Yes any future refactor will take place if and when we support rollback or showing more than one production / preview version in the dashboard.

**Pull request merged**